src/lib/Algoan.dto.ts

```diff
@@ -80,3 +88,28 @@ export interface PatchApplicationDTO {
   skipGDPF?: boolean;
   externalErrors?: ExternalError[];
 }
+
+/**
+ * POST /folders/:id/legal-documents DTO interface
+ */
+export interface PostLegalDocumentDTO {
+  category: LegalDocumentCategory;
+  holder: Holder;
+  required: boolean;
+  partnerId: string;
+  period: DocumentPeriod;
+  redirectUrl: string;
+  redirectUrlTTL: number;
+  rejectionCode: RejectionCode;
+  validFileTypes: LegalFileType[];
+}
+
+/**
+ * POST /folders/:id/legal-documents/:id/files DTO interface
+ */
+export interface PostLegalFileDTO {
+  file: string;
+  rejectionCode: number;
+  state: FileState;
+  type: LegalFileType;
+}
```

Comment:
Actually, `file` is a real media

```suggestion
export interface PostLegalFileDTO {
  file: ReadStream;
  rejectionCode?: number;
  state?: FileState;
  type?: LegalFileType;
}
```

Reply:
> In fact, the property file is optional so the request can be:

> * a multipart/form-data if the file is defined. Therefore, we need to create a form-data from the read stream
> * a classic JSON API if it is not defined. (Some of our partners can create files without the actual media, just to share extracted information from the file)

So here `file` should optional, right?

Reply:
In addition, I don't find `ReadStream` interface. Where can I find it?

Reply:
> In addition, I don't find `ReadStream` interface. Where can I find it?

Finally, I found it.

Reply:
I'm sorry I forgot to answer! It should be available in the @types/node lib right?

Reply:
Undireclty yes but I need to make this import: 
```ts
import { ReadStream } from "fs"
```